Frameworks/CoreGraphics/CGPath.mm

```diff
@@ -135,10 +135,10 @@
 
     // Only end the figure if we're requesting to actually close a figure, or if we have allowed the backing sink to dictate figure calls.
     STDMETHOD_(void, EndFigure)(D2D1_FIGURE_END figureEnd) {
-        if (m_allowsFigureEndExceptFinal && figureEnd != D2D1_FIGURE_END_CLOSED) {
+        if (m_allowsFigureEndExceptFinal > 0 && figureEnd != D2D1_FIGURE_END_CLOSED) {
             m_delayedEndFigure = true;
             m_delayedEndFigureClosure = figureEnd;
-        } else if (figureEnd == D2D1_FIGURE_END_CLOSED || m_allowsFigureCalls) {
+        } else if (figureEnd == D2D1_FIGURE_END_CLOSED || m_allowsFigureCalls > 0) {
```

Comment:
technically, evaluating `integer` as a logical expression folds to `integer != 0`